sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -449,7 +449,8 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
     Batch("DML rewrite", fixedPoint,
       RewriteDeleteFromTable,
       RewriteUpdateTable,
-      RewriteMergeIntoTable),
+      RewriteMergeIntoTable,
+      ResolveOutputRelation),
```

Comment:
can we add some comments to explain how these rules above can make the plan unresolved and we need to run `ResolveOutputRelation` to re-resolve it?

Reply:
fixed